### ISSUE TITLE
Outlook 2010-complete message body of a recieved message is not read automatically, when message opens in a separate window.

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3020]
**Reported by sumandogra on 2013-02-22 14:54**
Outlook 2010
NVDA does not read the complete message body automatically when the received message opens in a separate window.  It just reads the first two or three lines and does not read to the end of the message body automatically.
Steps to reproduce:
1.  Open a message in Outlook 2010 with ENTER.
2.  Message opens in a new separate window.
3.  NVDA automatically starts reading the message.
4.  But, on reaching the message body, it just reads the first two or three lines and does not read to the end of the message body automatically.

**Comment by leonardder** [comment]
@sumandogra: Can you still reproduce this? If so:

1. Is the message being read inside or outside browse mode?
2. If the former, do you have automatic say all on page load enabled in your browse mode settings?

**Comment by Adriani90** [comment]
@leonardder I can reproduce this issue in MS Outlook 2016 with NVDA 2018.4 Beta II and Windows 10 1809 Update.
Even if automatic say all when page loads is enabled, NVDA reads only the first three lines or so after pressing enter on an e-mail in outlook.
This issue occurs in focus mode and in browse mode.

**Comment by Adriani90** [comment]
Update with NVDA alpha-20097,6a573cd9, in Outlook 365 only the first line is read automatically when opening a mail message.

**Comment by Dennisl123** [comment]
It should be configerable if NVDA automatically reads the body of the email or not.

**Comment by Adriani90** [comment]
This woked some years ago, you had to create a profile for outlook and enable “read entire page after load” in browse mode settings. Then NVDA reported the whole message body. When disabling “read entire page after load”, NVDA reported only the first line. But this does not work anymore, at least not in Office 365.

 

 

 

Von: Dennisl123 <notifications@github.com> 
Gesendet: Mittwoch, 13. Mai 2020 22:28
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Comment <comment@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Outlook 2010-complete message body of a recieved message is not read automatically, when message opens in a separate window. (#3020)

 

It should be configerable if NVDA automatically reads the body of the email or not.

**Comment by Dennisl123** [comment]
The option should be  configurable by the user if the email body is read.

**Comment by josephsl** [comment]
Hi,

I guess say all handler isn't being engaged for this specific issue. By the way, this issue is a bit different than #11157 proposal.

Thanks.

**Comment by DonRichmond** [comment]
I solved my own problem of the message not being read by Preferences>Settings>Advanced and selecting the options to use Microsoft UI Automation. Now NVDA works the same in eM Cliient and Outlook 365.

**Comment by Dennisl123** [comment]
There needs to be a way to turn off the automatic reading of messages.

**Comment by Qchristensen** [comment]
Just had a user report this and I can replicate:
Using :

NVDA 2021.3 
 
Windows 10 (64-bit) Version: 21H2 (2009), Build: 19044.1348 
 
Office 365 (64-bit) Version: 16.0.14701.20206 

With "Say all on page load" in NVDA"s browse mode options checked, NVDA only reads the first line of messages in Outlook.

The same behaviour was present in NVDA 2021.2, but this worked in NVDA 2021.1

This issue appears to have evolved through possibly several different issues (since it was first reported in 2013, but as I am experiencing it, appears to be new since NVDA 2021.2).

**Comment by CyrilleB79** [comment]
@sumandogra, @Adriani90, @Qchristensen: can any of you reproduce the issue?

If I read correctly, the issue can be rephrased as follows:
After having ensured that "Enable browse mode on page load" is checked in Browse mode settings, open an Outlook message in a new window. Only the first line(s) is (are) read, while it is expected that the whole message be read in say all. Let me know if my understanding is not correct.

On my end, I am not able to reproduce with Outlook 2021 LTSC.

@Dennisl123's insisting requests are off-topic in this issue and should already covered by "Automatic Say All on page load" disabled, in a dedicated Outlook profile if needed. If this solution is not enough, it should rather be discussed in #15310 that has been opened since then; thanks Dennis!

**Comment by Dennisl123** [comment]
If say all is turned off it doesn’t solve the issue.  This needs fixed.  any help is appreciated.

**Comment by CyrilleB79** [comment]
@Dennisl123 wrote:
> If say all is turned off it doesn’t solve the issue.  This needs fixed.  any help is appreciated.

Of course if say all on page load the issue is not solved at all. Indeed, the desired behaviour in this issue is that say all work as expeted when a message is open!

Any request related to a use case with "Automatic Say All on page load" disabled is off topic in this issue. Please read the initial description, or let me know if I am wrong.

**Comment by seanbudd** [comment]
Closing as won't fix. This is working in Modern Outlook
